### ISSUE TITLE
[Load Balancer] Internal balancer with SSL for public DNS but only internal?

### ISSUE THREAD

**Issue by yatesco** [rancher/rancher#10504]
**Rancher versions:**
rancher/server:latest
rancher/agent:1.2.7

**Infrastructure Stack versions:**
healthcheck:v0.3.3
ipsec:0.13.5
network-services:0.9.5/0.7.18
scheduler:0.8.2

**Docker version: (`docker version`,`docker info` preferred)**
```
Containers: 20
 Running: 18
 Paused: 0
 Stopped: 2
Images: 23
Server Version: 17.09.1-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 06b9cb35161009dcb7123345749fef02f7cea8e0
runc version: 3f2f8b84a77f73d38244dd690525642a72156c64
init version: 949e6fa
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.4.0-103-generic
Operating System: Ubuntu 16.04.3 LTS
OSType: linux
Architecture: x86_64
CPUs: 40
Total Memory: 188.8GiB
Name: rincewind
ID: 5EOV:7PXQ:K3VU:2PUM:UHUK:6OUW:N3SN:A6PR:R73Y:ONKV:UEKX:NFQ5
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: No swap limit support
```

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
```
NAME="Ubuntu"
VERSION="16.04.3 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.3 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial
```
`uname -r` = `4.4.0-103-generic`

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**
Bare
**Setup details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
Single rancher server with a single host
**Environment Template: (Cattle/Kubernetes/Swarm/Mesos)**
Cattle
**Steps to Reproduce:**
- create a public DNS record for my.domain to the public IP of the server
- create a service running nginx without a public port exposing 80
- create a loadbalancer on nginx which maps the internal HTTP/S for my.domain hostname to the nginx service
**Results:**
The nginx service should not be accessible when accessed outside of the rancher service. It should be accessible when VPNed into the rancher network.

In other words, I want to provide publically accessible DNS records (because VPN, Rancher's DNS and macOS isn't a happy combination) but only provide access to them from within the Rancher network. 

Another approach I might have used previously was limiting with firewalls but I can't do that as other 80/443 traffic should be globally accessible. 

Finally, the other approach I might try is reserving the internal private rancher IPs and have the public DNS map to those internal IPs.

Thoughts?

Thanks!

**Comment by superseb** [comment]
VPNed into the rancher network meaning you are in the same range as the hosts running the `rancher/agent`? You can let the LB bind to a specific IP, then the `iptables` rule will only be made for that IP, and the LB will only be reachable on the IP you specify. If this is not the IP the host is registered with, you have to let Rancher know it's there by using scheduler IPs (http://rancher.com/docs/rancher/v1.6/en/rancher-services/scheduler/#multiple-ips)

**Comment by yatesco** [comment]
Apologies for the delay - yes, I am VPNed into a docker running on Rancher (the community OpenVPN one). I will have a look at that documentation - thanks!

**Comment by loganhz** [comment]
Please let us know in a comment if you can still reproduce the issue, and we'll reopen it
